java.d/src/main/java/org/firehol/netdata/Main.java

```diff
@@ -30,21 +30,55 @@
 import org.firehol.netdata.plugin.Printer;
 import org.firehol.netdata.plugin.configuration.ConfigurationService;
 import org.firehol.netdata.utils.LoggingUtils;
+import org.firehol.netdata.utils.NetdataLevel;
+import org.firehol.netdata.utils.StringUtils;
 
+/** @see <a href="https://github.com/firehol/netdata/wiki/External-Plugins#native-netdata-plugin-api">External-Plugins: native netdata plugin API</a> */
 public final class Main {
 	private static final Logger log = Logger.getLogger("org.firehol.netdata.plugin");
 
+
+	/** @see <a href="https://github.com/firehol/netdata/wiki/Tracing-Options">Tracing Options</a> */
+	private static final String NETDATA_DEBUG_FLAGS = "NETDATA_DEBUG_FLAGS";
+
+	/** @see <a href="https://github.com/firehol/netdata/blob/master/src/log.h">log.h</a> */
+	private static final long D_PLUGINSD = 0x0000000000000800;
+
 	private static List<Module> modules = Collections.emptyList();
 
 	private Main() {
 	}
 
+	/** @see <a href="https://github.com/firehol/netdata/wiki/External-Plugins#command-line-parameters">External-Plugins: command line parameters</a> */
 	public static void main(final String[] args) {
+		// allow increasing verbosity before reading configuration option
+		// org.firehol.netdata.plugin.configuration.schema.PluginDaemonConfiguration#logFullStackTraces
+		LoggingUtils.LOG_TRACES = isNetdataPluginDebugEnabled();
+
+		// handle updateEvery parameter provided by netdata
 		int updateEverySecond = getUpdateEveryInSecondsFomCommandLineFailFast(args);
+
 		configureModules();
 		new Plugin(updateEverySecond, modules).start();
 	}
 
+	/**
+	 * Chaecks whether the environment Netdata tracing flag for {@code plugins.d} is set.
+	 * 
+	 * @see <a href="https://github.com/firehol/netdata/wiki/Tracing-Options">Tracing Options</a>
+	 */
+	public static boolean isNetdataPluginDebugEnabled() {
```

Comment:
I would like to only have code in Main showing the main workflow. Please put this method at EnvironmentConfigurationService.